File: 品味Java的21種設計模式/dp_cpp/decorator/example4/ConcreteComponent.cpp
#include "ConcreteComponent.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace decorator
			{
				namespace example4
				{

					double ConcreteComponent::calcPrize(std::string user, Date *begin, Date *end)
					{
						//只是一个默认的实现，默认没有奖金
						return 0;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/decorator/example4/ConcreteComponent.java
package cn.javass.dp.decorator.example4;

import java.util.Date;

/**
 * 基本的实现计算奖金的类，也是被装饰器装饰的对象
 */
public class ConcreteComponent extends Component {

    public double calcPrize(String user, Date begin, Date end) {
        //只是一个默认的实现，默认没有奖金
        return 0;
    }
}
